### PR TITLE
Add JsonSchema fields

### LINKED ISSUE #60: Add missing properties from jsonschema
- [ ] [$id](https://json-schema.org/understanding-json-schema/structuring.html#id) 
- [ ] [anyOf](https://json-schema.org/understanding-json-schema/reference/combining.html#anyof)
- [ ] [const](https://github.com/softwaremill/sttp-apispec/issues/56
- [ ] [unevaluatedProperties](https://json-schema.org/understanding-json-schema/reference/object.html#unevaluated-properties)
- [ ] [dependentRequired](https://json-schema.org/understanding-json-schema/reference/conditionals.html#id4)
- [ ] [dependentSchemas](https://json-schema.org/understanding-json-schema/reference/conditionals.html#dependentschemas)

### PR DESCRIPTION
**Pull request by hamnis**
- Move ResourcePlatform to test-helper sbt-project
- Add tests to jsonschema-circe

Fixes #60 and #56

### CODE REVIEW

**Comment by hamnis** [comment]
mima fails as expected

**Comment by adamw** [comment]
cc @kciesielski I think that's related to your recent work on schemas

**Comment by kciesielski** [comment]
Thank you for the contribution, @hamnis! OpenAPI uses JSON Schema with some differences, for example `$id` and `const` fields are [unsupported](https://swagger.io/docs/specification/data-models/keywords/). Breaking mima by adding these fields would cause runtime errors for `.copy` calls on `Schema` if they are not added to the very end of the constructor. However, if you could move them after the `extensions` field, there should be no issues, at least from what I've quickly checked in my tests. Mima still fails, but `copy` works in the runtime, so I don't see any more risks, unless you could point some other possible issues I can't think of now?

**Comment by hamnis** [comment]
@kciesielski i have updated with the requested changes, Please have a look again.

File: circe-testutils/src/main/scalajs/sttp/apispec/test/ResourcePlatform.scala [hunk 1]
```diff
@@ -1,10 +1,11 @@
-package sttp.apispec.openapi.circe
+package sttp.apispec.test
 
 import io.circe.{Json, Error}
 import io.circe.parser.decode
 
 trait ResourcePlatform {
-  def rscPath(path: String): String = "openapi-circe/src/test/resources" + path
+  def basedir: String
```
Comment (adamw) [inline circe-testutils/src/main/scalajs/sttp/apispec/test/ResourcePlatform.scala:7]: I think I'd add some comments on what this should point to
Plus `rscPath` stands for `resourcesPath`? A bit more elaborate name could help as well :)

File: jsonschema-circe/src/test/scala/sttp/apispec/RoundTripTest.scala [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package sttp.apispec
+
+import org.scalatest.funsuite.AnyFunSuite
+import sttp.apispec.circe._
+import sttp.apispec.test._
+import io.circe.syntax._
+import io.circe.Decoder
+
+class RoundTripTest extends AnyFunSuite with ResourcePlatform {
+  val basedir = "jsonschema-circe"
```
Comment (adamw) [inline jsonschema-circe/src/test/scala/sttp/apispec/RoundTripTest.scala:10]: [minor] maybe add `override` here

**Review by kciesielski — APPROVED** [review]

**Comment by adamw** [comment]
@ahjohannessen a tapir release compatible with the above follows shortly - assuming you also have tapir in the classpath.

**Comment by ahjohannessen** [comment]
@adamw Yep 👍

**Pull request merged**